flake.nix

```diff
@@ -18,117 +18,122 @@
+                      dependencies = { llvm-config = llvmPackages.llvm.dev; };
+                      configureFlags = [ "--ghc-option=-optl=-lLLVM" ];
+                      extraLibraries = [ llvmPackages.llvm.dev ];
+                    };
 
-                  algebraic-graphs = dontCheck (mkCabal2nix {
-                    name = "algebraic-graphs";
-                    source = inputs.alga;
-                    doLibraryProfiling = enableLibraryProfiling;
-                  });
+                    algebraic-graphs = dontCheck (mkCabal2nix {
+                      name = "algebraic-graphs";
+                      source = inputs.alga;
+                      doLibraryProfiling = enableLibraryProfiling;
+                    });
 
-                  dependent-hashmap =
-                    unmarkBroken (dontCheck hp.dependent-hashmap);
+                    dependent-hashmap =
+                      unmarkBroken (dontCheck hp.dependent-hashmap);
 
-                  diagnose =
-                    hf.callCabal2nixWithOptions "diagnose" (inputs.diagnose)
-                    "-fmegaparsec-compat" { };
+                    diagnose =
+                      hf.callCabal2nixWithOptions "diagnose" (inputs.diagnose)
+                      "-fmegaparsec-compat" { };
 
-                  eclair-lang = mkCabal2nix {
-                    name = "eclair-lang";
-                    source = self;
-                    configureFlags = [ "--ghc-option=-optl=-lLLVM" ];
-                    haskellPackages = hf;
-                    extraLibraries = [ llvmPackages.llvm.dev ];
-                    overrideAttrs = _: {
-                      checkPhase = ''
-                        runHook preCheck
-                        DATALOG_DIR="${self}/cbits/" SOUFFLE_BIN="${final.souffle}/bin/souffle" ./Setup test
-                        runHook postCheck
-                      '';
+                    eclair-lang = mkCabal2nix {
+                      name = "eclair-lang";
+                      source = self;
+                      configureFlags = [ "--ghc-option=-optl=-lLLVM" ];
+                      haskellPackages = hf;
+                      extraLibraries = [ llvmPackages.llvm.dev ];
+                      doStaticLibraries = enableStaticLibraries;
+                      doSharedExecutables = disableSharedExecutables;
+                      overrideAttrs = old: {
+                        # /build/source/... is a virtual filesystem Nix uses
+                        # when building Eclair, so by adding it to the path we
+                        # can access the eclair executable during the tests.
+                        checkPhase = ''
+                          runHook preCheck
+                          DATALOG_DIR="${self}/cbits/" SOUFFLE_BIN="${final.souffle}/bin/souffle" ./Setup test
+                          PATH="${final.which}/bin:${final.souffle}/bin:/build/source/dist/build/eclair/:${llvmPackages.clang}/bin/:$PATH" ${final.lit}/bin/lit tests -v
```

Comment:
@luc-tielen I had to change this line to the following to get the `eclair` and `node` binaries to be found (thanks to @smunix for the tip :-)):
```bash
PATH="${final.which}/bin:${final.souffle}/bin:$PWD/dist/build/eclair:${llvmPackages.clang}/bin/:${pkgs.nodejs}/bin:$PATH" ${final.lit}/bin/lit tests -v
```

Reply:
Thanks, this helped. Also had to explicitly add 'node' when calling the `extract_snippet` script to make it work in `nix build`